### DISCUSSION TITLE
Is Github Workflow Action support C# DotNet Now?  (Not .NET Core, I'm asking .NET)

### DISCUSSION THREAD

**Discussion by Yang09701194** [community/community#26932] [Actions]
<p>Is Github Workflow Action support C# DotNet Now? (Not .NET Core, I’m asking .NET)</p>
<p>Is this file work on .NET  (not .NET Core)?</p>
<p><a href="https://github.com/actions/starter-workflows/blob/master/ci/dotnet-core.yml" rel="nofollow noopener">https://github.com/actions/starter-workflows/blob/master/ci/dotnet-core.yml</a></p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/yang09701194">@yang09701194</a> ,</p>
<p>Github actions supports any languages including C# DotNet. The file (dotnet-core.yaml) refered is a dotnet core workflow yaml file sample.</p>
<p>If you’d like to build .Net project, you can directly call ‘msbuild.exe’ which is installed on github hosted runner already, code sample as below:</p>
<pre><code>jobs:
  build:
    runs-on: windows-2019
    steps:
      - name: checkout
        uses: actions/checkout@v1
      - name: Build DotNET35
        run: |
           cd "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\"
           .\MSBuild.exe $Env:GITHUB_WORKSPACE\pathtoyoursolutionorproject
</code></pre>
<p>Or use below actions '<a href="https://github.com/warrenbuckley/Setup-MSBuild" rel="nofollow noopener">warrenbuckley/Setup-MSBuild’:</a></p>
<pre><code>jobs:
  build:
    runs-on: windows-latest

    steps:
    - uses: actions/checkout@master

    - name: Setup MSBuild.exe
      uses: warrenbuckley/Setup-MSBuild@v1

    - name: MSBuild
      working-directory: src
      run: msbuild MyProject.csproj
</code></pre>
<p>Hope it helps!</p>

**Comment by Yang09701194** [comment]
<p>Thank you, it works!</p>
<p>Add some description of my try for anyone new to this, the GITHUB_WORKSPACE of $Env is seen as repostiroy root,</p>
<p>so suppose my Id is <strong>Yang09701194</strong> , the repository name is  <strong>YangTestWorkflowAction</strong> ,</p>
<p>the sln file relative path is   <strong>Yang09701194/YangTestWorkflowAction/ConsoleApplication1/ConsoleApplication1.sln</strong></p>
<p>then write  <strong>$Env:GITHUB_WORKSPACE\ConsoleApplication1\ConsoleApplication1.sln</strong> works.</p>
<p>I have created an Example:</p>
<p>Build Succcess: <a href="https://github.com/Yang09701194/YangTestWorkflowAction/actions/runs/55459054" rel="nofollow noopener">https://github.com/Yang09701194/YangTestWorkflowAction/actions/runs/55459054</a></p>
<p>Workflow file: <a href="https://github.com/Yang09701194/YangTestWorkflowAction/actions/runs/55459054/workflow" rel="nofollow noopener">https://github.com/Yang09701194/YangTestWorkflowAction/actions/runs/55459054/workflow</a></p>

**Comment by AnkitKaushalTR** [comment]
<p>is there a way to use MSBuild version 12 in github actions?</p>
<p>tried using this in the workflow steps </p>
<p>run: |<br>
cd “C:\Program Files (x86)\MSBuild\12.0\Bin”</p>
<p>.\msbuild.exe</p>
<p>but it says it is not found in this location?</p>
